spec/mailers/notification_mailer_spec.rb

```diff
@@ -101,6 +107,140 @@
+
+      it { expect(email.subject).to eq("if me | #{who_triggered_event.name} created a group \"#{group.name}\"") }
+      it {expect(email.body.encoded).to match("Hi #{recipient.name},")}
+      it { expect(email.body.encoded).to match("<p>#{who_triggered_event.name} created a group \"#{group.name}\":</p><p><i>#{group.description}</i></p><p>To learn more and join, #{group_link}!</p><p>Please follow our #{code_of_conduct_link}!</p>") }
+    end
+
+    context 'when type is new_group_member' do
+      let(:type) { 'new_group_member' }
+
+      subject(:email) { NotificationMailer.notification_email(recipient, data) }
+
+      it { expect(email.subject).to eq("if me | #{who_triggered_event.name} joined your group \"#{group.name}\"") }
+      it {expect(email.body.encoded).to match("Hi #{recipient.name},")}
+      it { expect(email.body.encoded).to match("<p>To see \"#{group.name}\", #{group_link}!</p>") }
+    end
+
+    context 'when type is add_group_leader' do
+      let(:type) { 'add_group_leader' }
+
+      subject(:email) { NotificationMailer.notification_email(recipient, data) }
+
+      it { expect(email.subject).to eq("if me | #{who_triggered_event.name} became a leader of \" #{group.name}\"") }
+      it {expect(email.body.encoded).to match("Hi #{recipient.name},")}
+      it { expect(email.body.encoded).to match("<p>To see \"#{group.name}\", #{group_link}!</p>") }
+    end
+
+    context 'when type is add_group_leader and sender is recipient' do
+      let(:type) { 'add_group_leader' }
+
+      subject(:email) { NotificationMailer.notification_email(who_triggered_event, data) }
+
+      it { expect(email.subject).to eq("if me | You became a leader of \"#{group.name}\"") }
+      it {expect(email.body.encoded).to match("Hi #{who_triggered_event.name},")}
+      it { expect(email.body.encoded).to match("<p>To see \"#{group.name}\", #{group_link}!</p>") }
+    end
+
+    context 'when type is remove_group_leader' do
+      let(:type) { 'remove_group_leader' }
+
+      subject(:email) { NotificationMailer.notification_email(recipient, data) }
+
+      it { expect(email.subject).to eq("if me | #{who_triggered_event.name} is no longer a leader of \"#{group.name}\"") }
+      it {expect(email.body.encoded).to match("Hi #{recipient.name},")}
+      it { expect(email.body.encoded).to match("<p>To see \"#{group.name}\", #{group_link}!</p>") }
+    end
+
+    context 'when type is remove_group_leader and sender is recipient' do
+      let(:type) { 'remove_group_leader' }
+
+      subject(:email) { NotificationMailer.notification_email(who_triggered_event, data) }
+
+      it { expect(email.subject).to eq("if me | You are no longer a leader of \"#{group.name}\"") }
+      it {expect(email.body.encoded).to match("Hi #{who_triggered_event.name},")}
+      it { expect(email.body.encoded).to match("<p>To see \"#{group.name}\", #{group_link}!</p>") }
+    end
+
+    describe 'when meeting notification' do
```

Comment:
what are we describing here?..feels more like a context

Reply:
I think you're right. I'll take a look and update it.